botocore/stub.py

```diff
@@ -265,7 +314,21 @@ def _get_response_handler(self, model, params, **kwargs):
     def _assert_expected_params(self, model, params, **kwargs):
         self._assert_expected_call_order(model, params)
         expected_params = self._queue[0]['expected_params']
-        if expected_params is not None and params != expected_params:
+        if expected_params is None:
+            return
+
+        # Validate the parameters are equal
+        for param, value in expected_params.items():
+            if value is ANY:
+                continue
+            elif expected_params[param] != params.get(param, None):
```

Comment:
I think it's useful to distinguish between explicitly set values versus missing values.  I'd expect something like this to trigger an assertion error:

`{"foo": None} == {}`